api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java

```diff
@@ -151,12 +153,14 @@ public boolean hasUserSignedACurrentDucc(DbUser targetUser) {
 
   @Override
   public boolean isModuleCompliant(DbUser dbUser, DbAccessModuleName accessModuleName) {
-    DbAccessModule dbAccessModule =
-        getDbAccessModuleOrThrow(dbAccessModulesProvider.get(), accessModuleName);
     // if the module is not required, the user is always compliant
-    if (!isModuleRequiredInEnvironment(storageAccessModuleToClient(dbAccessModule.getName()))) {
+    if (!isModuleRequiredInEnvironment(
```

Comment:
slight logic change: now checks for presence in environment before checking the DB